contributors/design-proposals/resource-class.md

```diff
@@ -0,0 +1,209 @@
+  * If I request a resource `nvidia.gpu.high.mem` for my pod, any 'nvidia-gpu' type device which has memory greater than or equal to 'X' GB, should be able to satisfy this request, independent of other device capabilities such as 'version' or 'nvlink locality' etc.
+  * Similarly, if I request a resource `fast.nic`, any nic device with speed greater that 'B' gbps should be able to meet the request.
+  *  I want a rich metadata selection interface where operators like ‘Less Than’, ‘Greater Than’ and ‘Equal To’, are supported on the compute resource metadata.
+
+## Objectives
+
+1. Define and add support in the API for a new type, *Resource Class*.
+2. Add support for *Resource Class* in the scheduler.
+
+## Non Objectives
+1. Discovery, advertisement, allocation/deallocation of devices is expected to be addressed by [device plugin proposal](https://github.com/kubernetes/community/pull/695/files)
+
+## Resource Class
+*Resource Class* is a new type, objects of which provides abstraction over [Devices](https://github.com/RenaudWasTaken/community/blob/a7762d8fa80b9a805dbaa7deb510e95128905148/contributors/design-proposals/device-plugin.md#resourcetype). A *Resource Class* object selects devices using `matchExpressions`, a list of (operator, key, value). A *Resource Class* object selects a device if atleast one of the `matchExpressions` matches with device details. Within a matchExpression, all the (operator,key,value) are ANDed together to evaluate the result. 
+
+YAML example:
+```yaml
+kind: ResourceClass
+metadata:
+  name: nvidia.high.mem
+spec:
+  resourceSelector:
+    -
+      matchExpressions:
+        -
+          key: "type"
+          operator: "In"
+          values:
+            - "nvidia-gpu"
+          key: "memory"
+          operator: "Gt"
+          values:
+            - "30G"
+```
+
+## API Changes
+### ResourceClass
+
+Internal representation of *Resource Class*:
+
+```golang
+// +nonNamespaced=true
+// +genclient=true
+
+type ResourceClass struct {
+        metav1.TypeMeta
+        metav1.ObjectMeta
+        // Spec defines resources required
+        Spec ResourceClassSpec
+        // +optional
+        Status ResourceClassStatus
+}
+// Spec defines resources required
+type ResourceClassSpec struct {
+        // Resource Selector selects resources
+        ResourceSelector []ResourcePropertySelector
+}
+
+// A null or empty selector matches no resources
+type ResourcePropertySelector struct {
+        // A list of resource/device selector requirements. ANDed from each ResourceSelectorRequirement
+        MatchExpressions []ResourceSelectorRequirement
+}
+
+// A resource selector requirement is a selector that contains values, a key, and an operator
+// that relates the key and values
+type ResourceSelectorRequirement struct {
+        // The label key that the selector applies to
+        // +patchMergeKey=key
+        // +patchStrategy=merge
+        Key string
+        // Example 0.1, intel etc
```

Comment:
these are example of `Values`. For example if `Key` is 'version', `Values` can have '0.1'  OR `Key` is 'vendor', `Values` can have 'intel'. I will make it more clear.